app/controllers/api/groups_controller.rb

```diff
@@ -142,11 +142,33 @@ def update_marks
 
     def set_mark_by_criteria(criteria, mark_to_change)
       if criteria.is_a?(FlexibleCriterion)
-        mark_to_change.mark = params[criteria.flexible_criterion_name].to_f
+        if params[criteria.flexible_criterion_name] == "nil"
+          mark_to_change.mark = nil
+        else
+          mark_to_change.mark = params[criteria.flexible_criterion_name].to_f
+        end
       else
-        mark_to_change.mark = params[criteria.rubric_criterion_name]
+        if params[criteria.rubric_criterion_name] == "nil"
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.